src/strategy/weekly.rs

```diff
@@ -0,0 +1,122 @@
+//! Strategy for weekly weekly updates.
+
+use crate::config::inputs;
+use crate::weekly::{utils, DailyWindow, WeeklyCalendar};
+use failure::{Error, Fallible};
+use futures::future;
+use futures::prelude::*;
+use log::trace;
+use serde::Serialize;
+use std::pin::Pin;
+use std::time::Duration;
+
+/// Strategy for weekly weekly updates.
+#[derive(Clone, Debug, Default, Serialize)]
+pub(crate) struct StrategyWeekly {
+    /// Whitelisted time windows during which updates are allowed.
+    allowed_intervals: WeeklyCalendar,
+}
+
+impl StrategyWeekly {
+    /// Build a new Weekly strategy.
+    pub fn new(cfg: inputs::UpdateInput) -> Fallible<Self> {
+        let mut intervals = Vec::with_capacity(cfg.weekly.intervals.len());
+
+        for entry in cfg.weekly.intervals {
+            let weekday = utils::weekday_from_string(&entry.start_day)?;
+            let start = utils::time_from_string(&entry.start_time)?;
+            let length = Duration::from_secs(u64::from(entry.length_minutes).saturating_mul(60));
+            let windows = DailyWindow::parse_timespan(weekday, start.0, start.1, length)?;
+            intervals.extend(windows);
+        }
+
+        let calendar = WeeklyCalendar::new(intervals);
+        match calendar.total_length_minutes() {
+            0 => failure::bail!("zero-length weekly updates window"),
+            n => log::trace!("weekly updates, total length: {} minutes", n),
```

Comment:
Hmm, wouldn't it be more useful to report the non-overlapping total number of minutes here?

Reply:
Absolutely yes, but I'd need to implement that in the underlying library first (I have it on my radar). This is a trace-log not seen by default and mostly meant for debugging.
Once we have the correct number, it should be better exposed via metrics or such.

Reply:
For reference, https://github.com/main--/rust-intervaltree/issues/7 is in 0.2.6. I've left a TODO to compute the proper length measure here later on.